python/tritonserver/_c/tritonserver_pybind.cc

```diff
@@ -1175,21 +1162,109 @@ class PyInferenceRequest
     ThrowIfError(TRITONSERVER_InferenceRequestSetDoubleParameter(
         triton_object_, key.c_str(), value));
   }
+
   void Cancel()
   {
     ThrowIfError(TRITONSERVER_InferenceRequestCancel(triton_object_));
   }
 
+  // Response management
+  void GetNextResponse(const py::object& py_future)
+  {
+    // Called exclusively from Python, so GIL is always held.
+    std::lock_guard lock(response_mu_);
 
- public:
-  std::unique_ptr<CallbackResource> request_callback_resource_{nullptr};
+    // There are two cases:
+    // 1. The backend is faster than the frontend:
```

Comment:
```suggestion
    // 1. Responses have been generated and queued before GetNextResponse() has been called
    // 2. GetNextResponse() is called before responses have been generated and queued. 
```

I would remove the notion of "faster" and "slower" - I think that can be confusing here - it's not necessarily about speed of execution (i.e. applicaiton may be purposefully not reading, handling something else, etc.) - faster and slower makes it seem like a rate matching issue - but's really just whether or not there are items available when GetNextResponse is called ....

Reply:
also change the description to match the if / else order, so:

1. GetNextResponse is called before / when no responses have been queued.

Reply:
sure, the comment is updated. [Update comments related to GetNextResponse() and AddNextResponse()](https://github.com/triton-inference-server/core/pull/426/commits/1d4f98df4530d1e0b75ee426f33ff80d035840c0)